src/bind.py

```diff
@@ -202,24 +211,29 @@ def update_zonefiles_and_reload(
                 encoding="utf-8",
             )
 
-            # Write zone files
-            zone_files: dict[str, str] = self._zones_to_files_content(zones)
-            for domain, content in zone_files.items():
-                pathlib.Path(tempdir, f"db.{domain}").write_text(content, encoding="utf-8")
+            if topology is not None and topology.is_current_unit_active:
+                # Write zone files
+                zone_files: dict[str, str] = self._zones_to_files_content(zones)
+                for domain, content in zone_files.items():
+                    pathlib.Path(tempdir, f"db.{domain}").write_text(content, encoding="utf-8")
 
             # Write the named.conf file
             pathlib.Path(tempdir, "named.conf.local").write_text(
-                self._generate_named_conf_local([z.domain for z in zones]), encoding="utf-8"
+                self._generate_named_conf_local([z.domain for z in zones], topology),
```

Comment:
What happens if topology is none here?

Reply:
the `_generate_named_conf_local()` function will test if it's `None` and just output the service zone.  
But I think that I would need to separate the generate_named_conf_local function in two in the future: one for the service zone and another for the requested zones themselves.